src/mcts/search.cc

```diff
@@ -140,6 +143,20 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
       uci_info.pv.push_back(iter.GetMove(flip));
       if (!iter.node()) break;  // Last edge was dangling, cannot continue.
     }
+    // For mate display offset certain scores by 20000 + length of pv (average
+    // mate). Length of mate is +1000 if winning certain score is based on a
+    // propagated TBHit. For root filtered TB search, use best_rank (mate +500)
+    // TODO: include proper mate scores in uci_info structure
+    if (params_.GetCertaintyPropagation() > 0) {
+      if (edge.IsCertain() && edge.GetEQ() != 0.0f)
+        uci_info.score =
+            edge.GetEQ() * (20000 + ((uci_info.pv.size() + 1) / 2) + 1 +
```

Comment:
As I commented otherwise, add proper mate reporting in uci_info, without need for 20000s.

Reply:
In his defense, in cases where the mate is provable, but not all variations have been checked (e.g., tablebase wins) it is normal chess engine behavior to output a score of 100 or 300 + static eval, or an otherwise similarly inflated eval. On the other hand, if all variations have been checked, I agree that a mate score should be properly reported via uci_info.